### PR TITLE
Adds the missing FIRActionCodeOperationRecoverEmail.

### PR DESCRIPTION
**Pull request by XiangtianDai**
Also allow in-app action code handling to cancel and leave the code in pasteboard.

### CODE REVIEW

File: AuthSamples/Sample/MainViewController.m [hunk 1]
```diff
@@ -843,8 +844,8 @@ - (BOOL)handleIncomingLinkWithURL:(NSURL *)URL {
     return YES;
   }
   if ([mode isEqualToString:kVerifyEmailAction]) {
-    [self showMessagePromptWithTitle:@"Verify Email"
-                             message:@"Proceed?"
+    [self showMessagePromptWithTitle:@"Verify Email?"
```
Comment (protocol86) [inline AuthSamples/Sample/MainViewController.m]: how about the message "Tap Ok to very email"

File: Firebase/Auth/Source/FIRAuth.m [hunk 2]
```diff
@@ -104,10 +104,21 @@
 static NSString *const kMissingEmailInvalidParameterExceptionReason =
     @"The email used to initiate password reset cannot be nil.";
 
+/** @var kPasswordResetRequestType
+    @brief The action code type value for resetting password in the checking action code response.
```
Comment (protocol86) [inline Firebase/Auth/Source/FIRAuth.m]: s/checking/check

Comment (XiangtianDai) [inline Firebase/Auth/Source/FIRAuth.m]: Done

File: AuthSamples/Sample/MainViewController.m (hunk 1 above)
Comment (XiangtianDai) [inline AuthSamples/Sample/MainViewController.m]: Done

**Pull request merged**